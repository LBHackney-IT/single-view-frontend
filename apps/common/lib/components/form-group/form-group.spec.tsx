import React from "react";

import { render } from "@hackney/mtfh-test-utils";
import { screen } from "@testing-library/react";

import { FormGroup, Input } from "..";

test("it renders correctly", () => {
  render(
    <FormGroup id="test" name="test" label="Test Label">
      <Input placeholder="Test Value" />
    </FormGroup>,
  );

  expect(screen.getByText("Test Label")).toBeInTheDocument();
  expect(screen.getByPlaceholderText("Test Value")).toBeInTheDocument();
});

test("it shows the hint if defined", () => {
  render(
    <FormGroup id="test" name="test" label="Test Label" hint="Test Hint">
      <Input placeholder="Test Value" />
    </FormGroup>,
  );

  expect(screen.getByText("Test Hint")).toBeInTheDocument();
});

test("it shows the error if defined", () => {
  render(
    <FormGroup id="test" name="test" label="Test Label" error="Error">
      <Input placeholder="Test Value" />
    </FormGroup>,
  );

  expect(screen.getByText("Error")).toBeInTheDocument();
});

test("it adds an astreix to the label if required", () => {
  const { container } = render(
    <FormGroup id="test" name="test" label="Test Label" required>
      <Input placeholder="Test Value" />
    </FormGroup>,
  );
  expect(screen.getByText(/Test Label/).textContent).toBe("Test Label*");
  expect(container).toMatchSnapshot();
});
