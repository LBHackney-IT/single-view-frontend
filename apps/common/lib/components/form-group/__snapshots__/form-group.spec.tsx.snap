// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it adds an astreix to the label if required 1`] = `
<div>
  <div
    class="govuk-form-group lbh-form-group"
    id="test"
  >
    <label
      class="govuk-label lbh-label"
      for="test-field"
    >
      Test Label
      <sup
        aria-hidden="true"
      >
        *
      </sup>
    </label>
    <input
      class="govuk-input lbh-input"
      id="test-field"
      name="test"
      placeholder="Test Value"
      required=""
    />
  </div>
</div>
`;
